workflow/controller/operator.go

```diff
@@ -1090,6 +1099,24 @@ func (woc *wfOperationCtx) shouldPrintPodSpec(node wfv1.NodeStatus) bool {
 		(woc.controller.Config.PodSpecLogStrategy.FailedPod && node.FailedOrError())
 }
 
+func (woc *wfOperationCtx) getProgress(pod *apiv1.Pod, node *wfv1.NodeStatus) wfv1.Progress {
+	progress := wfv1.Progress("0/1")
+	if node.Progress.IsValid() {
+		progress = node.Progress
+	}
+
+	if annotation, ok := pod.Annotations[common.AnnotationKeyProgress]; ok {
+		v, ok := wfv1.ParseProgress(annotation)
+		if ok {
+			progress = v
+		}
+	}
+	if node.Fulfilled() {
+		progress = progress.Complete()
```

Comment:
if the progress is specified as `50/100` does complete make it `100/100` or `1/1`?

Reply:
it marks it as `100/100`. This ensures the progress doesn't suddenly change from `99/100` to `1/1`, i.e. keeps the same scale.

Reply:
awesome